Analyzer/plugins/Analyzer.cc

```diff
@@ -1040,57 +1596,61 @@ void Analyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
           dPtMinBcg = dPt;
         }
       }
-    }
-    
+    }//end of isData condition
+    bool kept_after_mcMatching = true;
+
     if (!isData && closestGenIndex < 0 ) {
       if (debug_ > 4 ) LogPrint(MOD) << "min dr: " << dRMinGen << endl;
       if (debug_ > 4 ) LogPrint(MOD) << "min dPt: " << dPtMinBcg << endl;
       // dont look at events where we didnt find the gen canidate
       if (debug_ > 4 ) LogPrint(MOD) << "  >> Event where we didnt find the gen candidate";
       // 5-th bin of the error histo, didnt find the gen canidate
+
       if (trigInfo_> 0 && matchToHLTTrigger_ && dr_min_hlt_muon_inEvent < 0.15) {
         tuple->ErrorHisto->Fill(4.);
       }
-      continue;
+      // continue;
+      ErrorHisto_bin = 4;
+      kept_after_mcMatching = false;
```

Comment:
I think in the end we dont need this at all, but ok I'll take care of it

Reply:
I added this so that in the ntuples, we are saving all HSCP candidates, not just the ones matched to gen-level. I use the `kept_after_mcMatching` as a book keeping boolean, so that all of your histograms are not affected